projects/vdk-plugins/vdk-oracle/src/vdk/plugin/oracle/oracle_connection.py

```diff
@@ -14,24 +16,52 @@
 
 class OracleConnection(ManagedConnectionBase):
     def __init__(
-        self, user: str, password: str, connection_string: str, thick_mode: bool = True
+        self,
+        user: str,
+        password: str,
+        connection_string: str = None,
+        host=None,
+        port=1521,
+        sid=None,
+        thick_mode: bool = True,
+        thick_mode_lib_dir: Optional[str] = None,
     ):
         super().__init__(log)
         self._oracle_user = user
         self._oracle_password = password
+        self._host = host
+        self._port = port
+        self._sid = sid
         self._oracle_connection_string = connection_string
         self._thick_mode = thick_mode
+        self._thick_mode_lib_dir = thick_mode_lib_dir
 
-    def _connect(self) -> PEP249Connection:
+    def _connect(self) -> Connection:
```

Comment:
The abstract method signature is

```python
    @abstractmethod
    def _connect(self) -> PEP249Connection:
        """
        this method MUST be implemented by inheritors and should return PEP249 Connection object (unmanaged)
        """
        raise NotImplementedError
```

I'm aware of the discrepancy, I'm just curious if we should change the abstract method type hint as well.

Reply:
No, because each database provides its own class without a common abstract class actually though all follow a common specification(protocol) 

Our PEP249Connection is actually meant ot be a protocol as defined in https://peps.python.org/pep-0544/ which his used in cases of duck typing like those.

Changing the subclass is merely for convenience and clarity.